database/postgres-views/README.md

```diff
@@ -0,0 +1,266 @@
+# Why this research
+
+While working on [this card](https://github.com/packit/packit-service/issues/1328) I thought that a `sql view` could have been usefull to speed up queries and simplify the code.
+I took as a reference the code for the [usage statistics](https://github.com/packit/packit-service/blob/01cbaf4fc3eb1898a1bb9d73a638d218b7a9c8d0/packit_service/models.py#L812-L869).
+My goal was to avoid duplicating a field (as requested by step 3 in [this card](https://github.com/packit/packit-service/issues/1328)), speeding up the query and make the code more simple using `sql views`.
+Implicitly I was working also on [this card](https://github.com/packit/packit-service/issues/1986).
+
+# Materialized views
+
+My first thought was that `materialized views` could improve queries' performance.
+
+To create a `materialized view` -> **4097.513 ms**.
+
+```sql
+CREATE MATERIALIZED VIEW copr_target_builds_for_pull_requests AS
+    SELECT git_projects.project_url, git_projects.repo_name, copr_build_targets.build_id, pipelines.datetime
+    FROM git_projects
+    JOIN pull_requests ON git_projects.id = pull_requests.project_id
+    JOIN job_triggers ON job_triggers.trigger_id = pull_requests.id
+    JOIN pipelines ON pipelines.job_trigger_id = job_triggers.id
+    JOIN copr_build_groups ON copr_build_groups.id = pipelines.copr_build_group_id
+    JOIN copr_build_targets ON copr_build_targets.copr_build_group_id = copr_build_groups.id
+    ORDER BY git_projects.project_url, git_projects.instance_url;
+```
+
+---
+
+To refresh the `materialized view` -> **5526.618 ms**
+
+```
+SELECT * FROM copr_target_builds_for_pull_requests;
+```
+
+---
+
+To query the `materialized view` -> **904.686 ms**
+
+```
+SELECT * FROM copr_target_builds_for_pull_requests;
+```
+
+---
+
+To create a `normal view` -> **25.266 ms**
+
+```sql
+CREATE VIEW copr_target_builds_for_pull_requests AS
+    SELECT git_projects.project_url, git_projects.repo_name, copr_build_targets.build_id, pipelines.datetime
+    FROM git_projects
+    JOIN pull_requests ON git_projects.id = pull_requests.project_id
+    JOIN job_triggers ON job_triggers.trigger_id = pull_requests.id
+    JOIN pipelines ON pipelines.job_trigger_id = job_triggers.id
+    JOIN copr_build_groups ON copr_build_groups.id = pipelines.copr_build_group_id
+    JOIN copr_build_targets ON copr_build_targets.copr_build_group_id = copr_build_groups.id
+    ORDER BY git_projects.project_url, git_projects.instance_url;
+```
+
+---
+
+To query the `normal view` -> **2616.131 ms**
+
+```
+SELECT * FROM copr_target_builds_for_pull_requests;
+```
+
+---
+
+Since we have to refresh a `materialized view`, like this, every time we use it (the data it contains change too often) sadly **there is no improvement**.
```

Comment:
This would make sense only if we did not need the newest data and could do the refresh periodically e.g.once a day and that would make sense only for the usage statistics and not for the data we need real-time in dashboard jobs/pipelines. Agreed that it is probably not worth spending time on that.